src/Form/QuestionType/QuestionTypeDateAndTime.php

```diff
@@ -0,0 +1,220 @@
+
+    /**
+     * Get the placeholder used in the date picker
+     * when the current time option is selected
+     *
+     * @return string
+     */
+    abstract public function getCurrentTimePlaceholder(): string;
+
+    /**
+     * Get the default value for the question type
+     *
+     * @return string
+     */
+    abstract public function getDefaultValue(?Question $question): string;
+
+    /**
+     * Check if the default value is the current time
+     *
+     * @param ?Question $question
+     * @return string
+     */
+    public function isDefaultValueCurrentTime(?Question $question): bool
+    {
+        if ($question === null) {
+            return false;
+        }
+
+        return $question->getExtraDatas()['is_default_value_current_time'] ?? false;
+    }
+
+    #[Override]
+    public function renderAdministrationTemplate(
+        ?Question $question = null
+    ): string {
+        $template = <<<TWIG
+            <script>
+                $(document).ready(function() {
+                    // Check if the global variable exists
+                    if (typeof window.flatpickr_configs === 'undefined') {
+                        window.flatpickr_configs = {};
+                    }
+
+                    // Check if the flatpickr instance exists
+                    if (
+                        $('#default-value_{{ rand }}').get(0) === undefined
+                        || $('#default-value_{{ rand }}').get(0)._flatpickr === undefined
+                    ) {
+                        return;
+                    }
+
+                    // Save the flatpickr config in a global variable
+                    window.flatpickr_configs['#default-value_{{ rand }}'] = $('#default-value_{{ rand }}').get(0)._flatpickr.config;
+
+                    // Register a listener to reinit the flatpickr instance when the question type is changed
+                    if (window.is_flatpickr_question_type_changed_listener_registered === undefined) {
+                        window.is_flatpickr_question_type_changed_listener_registered = true;
+
+                        $(document).on('glpi-form-editor-question-type-changed', function(event, question, type) {
+                            Object.entries(window.flatpickr_configs).forEach(([id, config]) => {
+                                // Check if the id is in the question and if it's visible
+                                if (
+                                    question.find(id).length > 0
+                                    && question.find(id).is(":visible")
+                                ) {
+                                    // Remove the last input
+                                    $(id).find('input').last().remove();
+
+                                    // Reinit the flatpickr instance
+                                    $(id).flatpickr(config);
+                                }
+                            });
+                        });
```

Comment:
We must be careful to define this event handler only once to prevent useless execution (we don't want the same code to be ran 10 times if there are 10 date questions).

Maybe we could only add the handler if `$question` is `null` ?
This would mean the handler will only be initialized when the question template is first rendered.

Also, we could check that the parameter `type` is equal to our type so the code only run if the question is in fact a datetime.